app/services/ui_notifications/clean_expired.rb

```diff
@@ -0,0 +1,43 @@
+module UINotifications
+  class CleanExpired
+    def initialize(options = {})
+      if options[:blueprint].present?
+        blueprint = NotificationBlueprint.find_by_name(options[:blueprint])
+        raise(Foreman::Exception, 'blueprint name does not exist') if blueprint.nil?
+        @blueprint = blueprint.name
+      end
+      if options[:expired_at].is_a? String
+        parsed_time =  options[:expired_at].to_time
+        raise(Foreman::Exception, 'Expired time cannot be greater then now') if parsed_time > Time.now.utc
+      end
+      @expired_at = parsed_time || Time.now.utc
+      @group = options[:group]
+    end
+
+    def clean!
+      Foreman::Logging.logger('notifications').info("Removing all expired notifications")
+      to_delete = Notification.joins(:notification_blueprint)
+                      .where('notifications.expired_at < ?', @expired_at)
+                      .where(blueprint_query, @blueprint)
+                      .where(group_query, @group)
+      @deleted_count = to_delete.size
```

Comment:
AFAIR the following line will provide you with a number, so you dont need to query first.